payments-core/src/test/java/com/stripe/android/payments/bankaccount/ui/CollectBankAccountViewModelTest.kt

```diff
@@ -0,0 +1,165 @@
+package com.stripe.android.payments.bankaccount.ui
+
+import app.cash.turbine.test
+import com.google.common.truth.Truth.assertThat
+import com.stripe.android.core.Logger
+import com.stripe.android.model.BankConnectionsLinkedAccountSession
+import com.stripe.android.payments.bankaccount.CollectBankAccountConfiguration
+import com.stripe.android.payments.bankaccount.domain.AttachLinkAccountSession
+import com.stripe.android.payments.bankaccount.domain.CreateLinkAccountSession
+import com.stripe.android.payments.bankaccount.navigation.CollectBankAccountContract
+import com.stripe.android.payments.bankaccount.navigation.CollectBankAccountContract.Args.ForPaymentIntent
+import com.stripe.android.payments.bankaccount.navigation.CollectBankAccountContract.Args.ForSetupIntent
+import com.stripe.android.payments.bankaccount.navigation.CollectBankAccountResult.Failed
+import com.stripe.android.payments.bankaccount.ui.CollectBankAccountViewEffect.FinishWithResult
+import com.stripe.android.payments.bankaccount.ui.CollectBankAccountViewEffect.OpenConnectionsFlow
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.flow.MutableSharedFlow
+import kotlinx.coroutines.test.runTest
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.mockito.kotlin.doReturn
+import org.mockito.kotlin.mock
+import org.mockito.kotlin.stub
+import org.robolectric.RobolectricTestRunner
+
+@ExperimentalCoroutinesApi
+@RunWith(RobolectricTestRunner::class)
+class CollectBankAccountViewModelTest {
+
+    private val createLinkAccountSession: CreateLinkAccountSession = mock()
+    private val attachLinkAccountSession: AttachLinkAccountSession = mock()
+
+    private val publishableKey = "publishable_key"
+    private val clientSecret = "client_secret"
+    private val name = "name"
+    private val email = "email"
+    private val linkedAccountSession = BankConnectionsLinkedAccountSession(
+        clientSecret = "las_client_secret",
+        id = "las_id"
+    )
+
+    @Test
+    fun `init - when createLinkAccountSession succeeds for PI, opens connection flow`() = runTest {
+        val viewEffect = MutableSharedFlow<CollectBankAccountViewEffect>()
+        viewEffect.test {
+            // Given
+            givenCreateAccountSessionForPaymentIntentReturns(Result.success(linkedAccountSession))
+
+            // When
+            buildViewModel(viewEffect, paymentIntentConfiguration())
```

Comment:
this is an example of starting to observe a flow before the viewmodel initializes.